packages/@aws-cdk/aws-codeartifact/lib/validation.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Throws an error if a value is defined and not an integer or not in a range.
+ */
+export function validate(name: string,
+  rule : { documentationLink: string, required?: boolean, pattern? : RegExp, minLength? : number, maxLength?: number },
+  value?: string) {
+
+  if (!value) {
+    if (rule.required) {
+      throw new Error(`'${name}' is required and cannot be empty`);
+    }
+    return;
+  }
+
+  if (rule.pattern && !rule.pattern.test(value)) {
+    throw new Error(`${name}: must match pattern ${rule.pattern.source}. Must match rules from ${rule.documentationLink}`);
```

Comment:
Adding the `documentationLink` here I don't think adds a ton of value, since the pattern is already in the message.